### PR TITLE
Merlin

### PR DESCRIPTION
**Pull request by JWvDronkelaar**
Django-axes is incompatible with locmemcache which is the default cache backend in our project (for now). To be able to use django-axes regardless we added a setting AXES_CACHE which allows us to set a different cache just for axes. This seems to have solved our issue for now, perhaps it is useful for other users as well.

I noticed an ongoing discussion about a flag to disable caching altogether. In my opinion this would be a nice option to have as well.

### CODE REVIEW

File: axes/utils.py [hunk 1]
```diff
@@ -1,17 +1,32 @@
...
-from axes.attempts import get_cache_key
 from axes.conf import settings
 from axes.models import AccessAttempt
 
 
+log = logging.getLogger(settings.AXES_LOGGER)
+
+
+def get_axes_cache():
+    axes_cache_name = settings.AXES_CACHE
```
Comment (aleksihakli) [inline axes/utils.py]: Could just be written as

```python
return caches[getattr(settings, 'AXES_CACHE', 'default')]
```

File: merlin.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+Make cache backend that django-axes uses configurable through a setting.
```
Comment (aleksihakli) [inline merlin.txt]: Is this intended to be included in the PR?

File: axes/utils.py [hunk 3]
```diff
@@ -1,17 +1,32 @@
...
+
 from platform import python_version
 from sys import platform
+
 if python_version() < '3.4' and platform == 'win32':
     import win_inet_pton
 from socket import inet_pton, AF_INET6, error
 
-from django.core.cache import cache
+from django.core.cache import cache, caches, InvalidCacheBackendError
```
Comment (aleksihakli) [inline axes/utils.py]: InvalidCacheBackendError is not used anywhere.

File: axes/utils.py [hunk 4]
```diff
@@ -1,17 +1,32 @@
+import logging
+
```
Comment (aleksihakli) [inline axes/utils.py]: Unnecessary line changes.

File: axes/utils.py [hunk 5]
```diff
@@ -1,17 +1,32 @@
...
 from platform import python_version
 from sys import platform
+
```
Comment (aleksihakli) [inline axes/utils.py]: Please include style changes in a separate PR.

File: axes/utils.py (hunk 1 above)
Comment (JWvDronkelaar) [reply axes/utils.py]: Agreed, will change.

File: merlin.txt (hunk 2 above)
Comment (JWvDronkelaar) [reply merlin.txt]: No it is not, I will remove it.

File: axes/utils.py (hunk 3 above)
Comment (JWvDronkelaar) [reply axes/utils.py]: Yes it is a remnant from an error I wanted to catch. I will remove it.

File: axes/utils.py (hunk 4 above)
Comment (JWvDronkelaar) [reply axes/utils.py]: I will revert it.

File: axes/utils.py (hunk 5 above)
Comment (JWvDronkelaar) [reply axes/utils.py]: Sure no problem.

**Comment by JWvDronkelaar** [comment]
Hi Aleksi! In short I agree with everything you suggested. I will make the changes you suggested. I don't know if I have the time to do much additional work on it on top of those changes.

That said it would be great if either cache choice or the disabling of cache altogether could become a part of this official library. It would make the requirements for our software a lot simpler/reliable (instead of having to install from a fork at a specific commit).

**Comment by aleksihakli** [comment]
I think this changeset looks nice now. What do you think @camilonova?

**Comment by aleksihakli** [comment]
@JWvDronkelaar this line should be changed to use the `AXES_CACHE` as well:

https://github.com/jazzband/django-axes/blob/master/axes/apps.py#L11

Then the checks for LocMemCache should function correctly.

**Comment by aleksihakli** [comment]
Also, could you add documentation to docs/ prior to merging about the `AXES_CACHE` setting and an example of its usage? Using DummyCache as the `AXES_CACHE` also effectively disables caching with this changeset.

**Comment by JWvDronkelaar** [comment]
Hello @aleksihakli . I have made the requested changes to the docs. The ready check now also takes the AXES_CACHE setting into account. For this I had to merge the Merlin branch with the updated master branch and I noticed the latest commit to master does not show here. Should I have commited it to the Merlin branch that I merged? I am new to forks and their workflow. If there is something I need to do to make this work for you please let me know.

**Comment by JWvDronkelaar** [comment]
@aleksihakli I have asked my colleague for advice and merged the master into the merlin branch instead which seems to give a better result. I hope this is the correct process, if not I'd love to hear from you on how to fix it.

**Comment by camilonova** [comment]
@aleksihakli looks good to me.

@JWvDronkelaar Great work pal! I would recommend to squash your commits into a single one for clarity purposes.

**Comment by aleksihakli** [comment]
Yeah, squash the commits and with `git rebase` and I think this is good for merging :)

**Comment by aleksihakli** [comment]
Hey, this has been now rebased and merged in #309 and you are the author of the cache work @JWvDronkelaar. I'm closing this PR since the content is already in master. Thank you!

**Pull request closed without merging**